### PR TITLE
Benchmark over a range of node counts

### PR DESCRIPTION
**Pull request by ryanslade**

### CODE REVIEW

**Review by tsenart — APPROVED** [review]
So, do the results look good enough?

File: rdv_test.go [hunk 1]
```diff
@@ -42,22 +42,34 @@ func TestLookupN_Lookup_Equivalence(t *testing.T) {
...
+		})
 	}
 }
 
 func BenchmarkLookupN(b *testing.B) {
 	b.ReportAllocs()
-	r := New(nodes(), xxhash.Sum64String)
-	for i := 0; i < b.N; i++ {
-		r.LookupN("github.com/foo/bar", 3)
+	b.ReportAllocs()
```
Comment (keegancsmith) [inline rdv_test.go]: nit: you are calling ReportAllocs twice

**Review by keegancsmith — APPROVED** [review]
Can you share the results of the benchmark?

**Comment by ryanslade** [comment]
> So, do the results look good enough?

Up to about 100 nodes it seems in line with our current hashing function, but it grows linearly with the number of nodes. I don't see us getting above 100 nodes any time soon so we should be fine. 

LookupN is a bit slower since it's doing some allocations but I'm sure we can work around that. We could return positions in the slice of nodes or possibly a subslice of the existing nodes slice so that we don't allocate a new array.

**Pull request merged**